truetype/truetype.go

```diff
@@ -306,6 +314,141 @@ func (f *Font) parseMaxp() error {
+			}
+			switch platformID {
+			case PLATFORM_UNICODE:
+				foundUnicode[nameID-1] = offset
+			case PLATFORM_MACINTOSH:
+				if languageID == 0 { // The language ID for English is zero.
+					foundAppleEnglish[nameID-1] = offset
+				} else if encodingID == 0 { // A zero encoding ID means Roman.
+					foundAppleRoman[nameID-1] = offset
+				}
+			case PLATFORM_MICROSOFT:
+				if foundMicrosoft[nameID-1] == 0 || (languageID&0x3FF) == 0x009 { // Prefer Microsoft fonts in English.
+					if encodingID == MICROSOFT_SYMBOL_CS || encodingID == MICROSOFT_UNICODE_CS || encodingID == MICROSOFT_UCS_4 {
+						foundMicrosoftEnglish = (languageID & 0x3FF) == 9
+						foundMicrosoft[nameID-1] = offset
+					}
+				}
+			}
+		}
+	}
+
+	foundApple = foundAppleRoman
+	if foundAppleEnglish[0] > 0 {
+		foundApple = foundAppleEnglish
+	}
+
+	nameRecord := nameInfo{}
+	// Should we decode UTF-16 strings?
+	switch {
+	case foundMicrosoft[0] > 0 && !(foundApple[0] > 0 && !foundMicrosoftEnglish):
+		nameOffset, styleOffset := u16(f.name, foundMicrosoft[0]+10)+stringOffset, u16(f.name, foundMicrosoft[1]+10)+stringOffset
+		nameLength, styleLength := u16(f.name, foundMicrosoft[0]+8), u16(f.name, foundMicrosoft[1]+8)
+
+		nameRecord.name = f.nameEntryInASCII(f.name[nameOffset:nameOffset+nameLength], true)
+		nameRecord.style = f.nameEntryInASCII(f.name[styleOffset:styleOffset+styleLength], true)
+	case foundApple[0] > 0:
+		nameOffset, styleOffset := u16(f.name, foundApple[0]+10)+stringOffset, u16(f.name, foundApple[1]+10)+stringOffset
+		nameLength, styleLength := u16(f.name, foundApple[0]+8), u16(f.name, foundApple[1]+8)
+
+		nameRecord.name = f.nameEntryInASCII(f.name[nameOffset:nameOffset+nameLength], false)
+		nameRecord.style = f.nameEntryInASCII(f.name[styleOffset:styleOffset+styleLength], false)
+	case foundUnicode[0] > 0:
+		nameOffset, styleOffset := u16(f.name, foundUnicode[0]+10)+stringOffset, u16(f.name, foundUnicode[1]+10)+stringOffset
+		nameLength, styleLength := u16(f.name, foundUnicode[0]+8), u16(f.name, foundUnicode[1]+8)
+
+		nameRecord.name = f.nameEntryInASCII(f.name[nameOffset:nameOffset+nameLength], true)
+		nameRecord.style = f.nameEntryInASCII(f.name[styleOffset:styleOffset+styleLength], true)
+	}
+
+	f.nameRecord = nameRecord
+
+	return nil
+}
+
+func (f *Font) nameEntryInASCII(b []byte, utf16 bool) string {
+	buf := &bytes.Buffer{}
+	if utf16 { // Equivalent to tt_name_ascii_from_utf16.
+		j := len(b)
+		if j&1 == 1 {
+			return ""
+		}
+
+		for i := 0; i < j; i += 2 {
+			el := u16(b, i)
+			if el == 0 {
+				continue
+			} else if el < 32 || el > 127 {
+				buf.WriteByte('?')
+			} else {
+				buf.WriteByte(byte(el))
+			}
+		}
+	} else { // Equivalent to tt_name_ascii_from_other.
+		for _, el := range b {
+			if el == 0 {
+				continue
+			} else if el < 32 || el > 127 {
+				buf.WriteByte('?')
+			} else {
+				buf.WriteByte(el)
+			}
+		}
+	}
+
+	return buf.String()
+}
+
+// Name returns the font's name.
+func (f *Font) Name() string {
```

Comment:
Instead of separate Name and Style methods, how about:

type NameID int

const (
  NameIDCopyright NameID = 0
  NameIDFontFamily NameID = 1
  NameIDFontSubfamily NameID = 2
  etc
)

as per the "Name Identifiers" table at https://developer.apple.com/fonts/TrueType-Reference-Manual/RM06/Chap6name.html

and then you'd have

func (f *Font) Name(id NameID) string